src/NuGet.Core/NuGet.Commands/RestoreCommand/CompatibilityChecker.cs

```diff
@@ -66,7 +66,7 @@ internal async Task<CompatibilityCheckResult> CheckAsync(
             if (packageSpec.RestoreMetadata?.ProjectStyle == ProjectStyle.DotnetToolReference)
             {
                 // Autoreferenced packages are allowed. Currently they're using Microsoft.NET.Platforms as an auto-ref package
-                if (packageSpec.GetAllPackageDependencies().Where(e => !e.AutoReferenced).Count() != 1)
+                if (packageSpec.GetAllPackageDependencies().Count(e => !e.AutoReferenced) != 1)
```

Comment:
Seems like we can enable this rule to capture all these issues: 

https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0120

Can we enable: 
```
dotnet_diagnostic.IDE0120.severity = warning
```

likely on line 149 in https://github.com/NuGet/NuGet.Client/blob/dev/.editorconfig.

Reply:
fwiw, enabling this does reveal some issues in NuGet.Frameworks\FrameworkReducer.cs for example.

Reply:
@nkolev92 done